code/modules/mob/living/simple_animal/friendly/trader.dm

```diff
@@ -57,9 +58,21 @@
 		"Oooooooo~!"
 	)
 
+/mob/living/simple_animal/hostile/retaliate/trader/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_CLICK, .proc/click_react)
```

Comment:
Why click and not attack_hand?

Reply:
I know you intentionally avoid adjacency so I know this part is intentional too, but why?

Reply:
we need proc calls regardless of range and the attack_hand is only sent when you're in range